packages/aws-cdk-lib/aws-cloudfront/lib/realtime-log-config.ts

```diff
@@ -0,0 +1,96 @@
+import { Construct } from 'constructs';
+import { CfnRealtimeLogConfig } from './cloudfront.generated';
+import { IResource, Resource } from '../../core';
+
+/**
+ * Endpoint data streams types.
+ */
+export enum DataStreamType {
+  KINESIS = 'Kinesis'
+}
+
+/**
+ * Contains information about the Amazon Kinesis data stream where you are sending real-time log data.
+ */
+export interface KinesisStreamConfig {
+  /**
+   * The Amazon Resource Name (ARN) of an AWS Identity and Access Management (IAM) role that CloudFront can use to send real-time log data to your Kinesis data stream.
+   */
+  readonly roleArn: string;
+  /**
+   * The Amazon Resource Name (ARN) of the Kinesis data stream where you are sending real-time log data.
+   */
+  readonly streamArn: string;
```

Comment:
These both should just be the respective interfaces. For example, `readonly role: IRole` and `readonly kinesisStream: IStream`